fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogStore.kt

```diff
@@ -0,0 +1,284 @@
+    initialState: TranslationsDialogState,
+    middlewares: List<Middleware<TranslationsDialogState, TranslationsDialogAction>> = emptyList(),
+) : Store<TranslationsDialogState, TranslationsDialogAction>(
+    initialState,
+    TranslationsDialogReducer::reduce,
+    middlewares,
+) {
+    init {
+        dispatch(TranslationsDialogAction.FetchSupportedLanguages)
+    }
+}
+
+/**
+ * The current state of the Translations bottom sheet dialog.
+ *
+ * @property isTranslated The page is currently translated.
+ * @property translationInProgress The page is currently attempting a translation.
+ * @property positiveButtonType Can be enabled,disabled or in progress.
+ * @property error An error that can occur during the translation process.
+ * @property dismissDialogState Whether the dialog bottom sheet should be dismissed.
+ * @property initialFrom Initial "from" language, based on the translation state and page state.
+ * @property initialTo Initial "to" language, based on the translation state and page state.
+ * @property fromLanguages Translation menu items to be shown in the translate from dropdown.
+ * @property toLanguages Translation menu items to be shown in the translate to dropdown.
+ * @property translatedPageTitle Title of the bottom sheet dialogue if the page was translated.
+ */
+data class TranslationsDialogState(
+    var isTranslated: Boolean = false,
+    val translationInProgress: Boolean = false,
+    val positiveButtonType: PositiveButtonType? = null,
+    val error: TranslationError? = null,
+    val dismissDialogState: DismissDialogState? = null,
+    val initialFrom: Language? = null,
+    val initialTo: Language? = null,
+    val fromLanguages: List<Language>? = null,
+    val toLanguages: List<Language>? = null,
+    val translatedPageTitle: String? = null,
+) : State
+
+/**
+ * Action to dispatch through the `TranslationsStore` to modify `TranslationsDialogState` through the reducer.
+ */
+sealed class TranslationsDialogAction : Action {
+    /**
+     * Invoked when the [TranslationsDialogStore] is added to the fragment.
+     */
+    object InitTranslationsDialog : TranslationsDialogAction()
+
+    /**
+     * When FetchSupportedLanguages is dispatched, an [TranslationsAction.OperationRequestedAction]
+     * will be dispatched to the [BrowserStore]
+     */
+    object FetchSupportedLanguages : TranslationsDialogAction()
+
+    /**
+     * Invoked when the user wants to translate a website.
+     */
+    object TranslateAction : TranslationsDialogAction()
+
+    /**
+     * Invoked when the user wants to restore the website to its original pre-translated content.
+     */
+    object RestoreTranslation : TranslationsDialogAction()
+
+    /**
+     * Invoked when a translation error occurs during the translation process.
+     */
+    data class UpdateTranslationError(
+        val translationError: TranslationError? = null,
+    ) : TranslationsDialogAction()
+
+    /**
+     *  Updates translate from languages list.
```

Comment:
```suggestion
     * Updates translate from languages list.
```
Might want to fix the spacing for all of these.

Reply:
DOne